grafana/resource_alert_notification.go

```diff
@@ -224,14 +224,10 @@ func makeAlertNotification(_ context.Context, d *schema.ResourceData) (*gapi.Ale
 	secureSettings := map[string]interface{}{}
 	for k, v := range d.Get("secure_settings").(map[string]interface{}) {
 		strVal, ok := v.(string)
-		switch {
-		case ok && strVal == "true":
-			secureSettings[k] = true
-		case ok && strVal == "false":
-			secureSettings[k] = false
-		default:
-			secureSettings[k] = v
+		if !ok {
+			return nil, errors.New("secure_settings must be a map of string")
 		}
+		secureSettings[k] = strVal
 	}
```

Comment:
I've tried different alternatives and `v` can always be casted into a string, no matter what we pass before (e.g. if we defined the value as a list then Terraform will identify it's an invalid value and fail before reaching this line). Because of this, we could have replaced all this code by the following and rely on Go's compiler to raise an error if it cannot cast the value to a string, however, this error is raised as a `panic`, so I'd rather have a more controlled error returned to the user.

```go
secureSettings := make(map[string]interface{})
for k, v := range d.Get("secure_settings").(map[string]interface{}) {
    secureSettings[k] = v.(string)
}
```